lightning/api_grpc.go

```diff
@@ -218,3 +225,509 @@ func (l *LndGrpcLightningAPI) GetChanInfo(ctx context.Context, chanID uint64) (*
 	ret := l.convertChan(resp)
 	return &ret, nil
 }
+
+// SubscribeForwards API
+func (l *LndGrpcLightningAPI) SubscribeForwards(ctx context.Context, since time.Time, batchSize uint16) (<-chan []ForwardingEvent, <-chan ErrorData) {
+	// We will first try obtaining ForwadingHistory and then move to SubscribeHtlc
+	const maxErrors = 5
+	sleepTime := 5 * time.Second
+
+	errorChan := make(chan ErrorData, 1)
+	outChan := make(chan []ForwardingEvent)
+
+	if batchSize == 0 {
+		batchSize = 50
+	}
+
+	errors := 0
+	req := &lnrpc.ForwardingHistoryRequest{
+		NumMaxEvents: uint32(batchSize),
+		IndexOffset:  uint32(0),
+		StartTime:    uint64(since.Unix()),
+	}
+
+	go func() {
+		var (
+			subscribeClient routerrpc.Router_SubscribeHtlcEventsClient
+		)
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				// Do nothing
+			}
+			resp, err := l.Client.ForwardingHistory(ctx, req)
+
+			if err != nil {
+				glog.Warningf("Error getting ForwadingHistory %v\n", err)
```

Comment:
why do we need logging if we are channeling errors on errorChan. Wouldnt these errors be logged on the other end ?